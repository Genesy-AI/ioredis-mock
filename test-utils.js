// Runs both a regular string suite and a buffer suite
export function runTwinSuite(commandName, cb) {
  const commands = [`${commandName}Buffer`, commandName]

  return commands.forEach(command =>
    command === commandName
      ? cb(command, (a, b) => a === b)
      : cb(
          command,
          (a, b) => a.equals(Buffer.from(b))
          /*
          // @TODO: detect when it happens so we can remove this
          if (Buffer.isBuffer(a)) return a.equals(Buffer.from(b))

          console.warn('a was expected to be a Buffer:', typeof a, { a, b })
          return a === b
          // */
        )
  )
}

// Util for calling describe.skip if the suite asserts Buffer outputs, which will fail in test:browser.js due to the polyfill
export const browserSafeDescribe = command =>
  process.env.IS_BROWSER && command.endsWith('Buffer')
    ? // eslint-disable-next-line no-undef
      describe.skip
    : // eslint-disable-next-line no-undef
      describe
