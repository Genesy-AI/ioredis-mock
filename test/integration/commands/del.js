import Redis from 'ioredis'

// eslint-disable-next-line import/no-relative-parent-imports
import { runTwinSuite } from '../../../test-utils'

runTwinSuite('del', command => {
  describe(command, () => {
    const redis = new Redis()

    afterAll(() => redis.disconnect())

    it('should delete passed in keys', async () => {
      await redis.set('deleteme', 'please')
      await redis.set('metoo', 'pretty please')

      expect(await redis[command]('deleteme', 'metoo')).toBe(2)
      expect(await redis.get('deleteme')).toBe(null)
      expect(await redis.get('metoo')).toBe(null)
    })
    it('return 0 if nothing is deleted', async () => {
      expect(await redis[command]('deleteme', 'metoo')).toBe(0)
    })

    it('return 0 if nothing is deleted', async () => {
      expect(await redis[command]('deleteme', 'metoo')).toBe(0)
    })

    it('throws if too few arguments', async () => {
      expect.assertions(1)

      try {
        await redis[command]()
      } catch (err) {
        expect(err.message).toMatchSnapshot()
      }
    })
  })
})
