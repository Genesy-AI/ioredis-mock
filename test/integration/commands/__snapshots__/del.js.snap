// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`del throws if too few arguments 1`] = `"ERR wrong number of arguments for 'del' command"`;

exports[`delBuffer throws if too few arguments 1`] = `"ERR wrong number of arguments for 'del' command"`;
